openshift_resources/oadp/data_protection_application.py

```diff
@@ -1,17 +1,516 @@
-import logging
-import wrapper_constants.resources as r
+# generated by datamodel-codegen:
+#   filename:  k8s_swagger.json
+#   timestamp: 2022-10-29T09:10:36+00:00
 
-from ocp_resources.resource import NamespacedResource
+from __future__ import annotations
+from datetime import datetime
+from enum import Enum
+from typing import Any, Dict, List, Optional
 
-logger = logging.getLogger(__name__)
+from pydantic import BaseModel, Field, conint, constr
+from kubernetes import config
+from openshift.dynamic import DynamicClient
+import resources.io.k8s.apimachinery.pkg.apis.meta.v1 as v1
 
+from openshift_resources.resource import BaseResource
 
-class DataProtectionApplication(NamespacedResource):
-    api_group = r.ApiGroups.OADP_API_GROUP.value
 
-    def wait_for_successful_reconciliation(self):
-        self.wait_for_condition(condition=r.Condition.CONDITION_RECONCILED.value, status=r.Status.SUC_RECONCILED.value)
+class CreationSecret(BaseModel):
+    key: str = Field(
+        ...,
+        description='The key of the secret to select from.  Must be a valid secret key.',
+    )
+    name: Optional[str] = Field(
+        None,
+        description='Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names TODO: Add other useful fields. apiVersion, kind, uid?',
+    )
+    optional: Optional[bool] = Field(
+        None, description='Specify whether the Secret or its key must be defined'
+    )
 
-    def wait_for_failed_reconciliation(self):
-        self.wait_for_condition(condition=r.Condition.CONDITION_RECONCILED.value, status=r.Status.FAILED_TO_RECONCILE.value)
 
+class Provider(Enum):
+    aws = 'aws'
+
+
+class CloudStorageSpec(BaseModel):
+    creationSecret: CreationSecret = Field(
+        ...,
+        description='creationSecret is the secret that is needed to be used while creating the bucket.',
+    )
+    enableSharedConfig: Optional[bool] = Field(
+        None,
+        description='enableSharedConfig enable the use of shared config loading for AWS Buckets',
+    )
+    name: str = Field(
+        ...,
+        description='name is the name requested for the bucket (aws, gcp) or container (azure)',
+    )
+    provider: Provider = Field(
+        ..., description='provider is the provider of the cloud storage'
+    )
+    region: Optional[str] = Field(
+        None,
+        description='region` for the bucket to be in, will be us-east-1 if not set.',
+    )
+    tags: Optional[Dict[str, str]] = Field(None, description='tags` for the bucket')
+
+
+class CloudStorageStatus(BaseModel):
+    lastSyncTimestamp: Optional[datetime] = None
+    name: str
+
+
+class CloudStorageRef(BaseModel):
+    name: Optional[str] = Field(
+        None,
+        description='Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names TODO: Add other useful fields. apiVersion, kind, uid?',
+    )
+
+
+class Credential(BaseModel):
```

Comment:
redundant, defined already in core.v1: https://github.com/oadp-qe/openshift-resources/blob/main/src/resources/io/k8s/api/core/v1.py#L1342